message_set.go

```diff
@@ -48,9 +48,11 @@ func (msb *MessageBlock) decode(pd packetDecoder) (err error) {
 type MessageSet struct {
 	PartialTrailingMessage bool // whether the set on the wire contained an incomplete trailing MessageBlock
 	Messages               []*MessageBlock
+	UnderlyingMessageCount int // in case of compressed messages wraps in a single fake "message"
```

Comment:
I'm not sure I understand why this is needed isn't it always just equal to `len(Messages)`?